packages/vite/src/node/server/index.ts

```diff
@@ -531,8 +531,17 @@ async function startServer(
 
   const options = server.config.server || {}
   let port = inlinePort || options.port || 3000
-  let hostname = options.host || 'localhost'
-  if (hostname === '0.0.0.0') hostname = 'localhost'
+  let hostname: string | undefined
+  if (options.host === undefined || options.host === 'localhost') {
+    // Use a secure default
+    hostname = '127.0.0.1'
+  } else if (options.host === true) {
+    // probably passed --host in the CLI, without arguments
+    hostname = undefined // undefined typically means 0.0.0.0 or :: (listen on all IPs)
+  } else {
+    hostname = options.host as string
```

Comment:
**question:** Is this also the case if `options.host` is set to `false`?
Cause I see that `host` is of type `string | boolean | undefined`

Reply:
@patak-js as I mentioned here, what is with the `options.host` being `false` case?

![image](https://user-images.githubusercontent.com/7195563/117107363-c0aece00-ad81-11eb-88cf-2c30cf6d3348.png)

The `as string` just hides the TypeScript safeness here ⚠️